llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -273,8 +274,27 @@ VPInstruction::VPInstruction(unsigned Opcode,
   assert(isFPMathOp() && "this op can't take fast-math flags");
 }
 
-Value *VPInstruction::generateInstruction(VPTransformState &State,
-                                          unsigned Part) {
+bool VPInstruction::doesGenerateScalars() const {
+  return Opcode == VPInstruction::PtrAdd;
```

Comment:
This is a bit confusing: PtrAdd is (currently) the only VPInstruction that may generate multiple scalars - per lane, or a single scalar - per lane zero only, but there are other VPInstructions that generate a single scalar - per part.

Perhaps `doesGeneratePerLane()` would be more accurate, or `doesGeneratePerAllLanes()`? See below.

Reply:
Updated to use `doesGeneratePerAllLanes`.